en/docs/learn/jarm-for-oauth2.md

```diff
@@ -0,0 +1,162 @@
+# JWT-Secured Authorization Response Mode (JARM) for OAuth 2.0
+
+The [JWT-Secured Authorization Response Mode for OAuth 2.0 (JARM) specification](https://bitbucket.org/openid/fapi/src/master/oauth-v2-jarm.md) defines new JWT-based modes to encode OAuth2 authorization responses. These new modes allow clients to request authorization response parameters and additional data in JWT format.
+
+!!! info
+    This is only available as a WSO2 Update from WSO2 Identity Server update level **5.10.216** onwards. See the instructions on [updating WSO2 products](https://updates.docs.wso2.com/en/latest/).
+
+## Authorization flow
+
+Given below is a sample authorization request sent to the authorization endpoint of WSO2 Identity Server.
+
+```bash
+https://<IS_HOME>/oauth2/authorize?prompt=login&scope=openid&redirect_uri=https://<CLIENT-HOST>/redirects/redirect1&client_id=<CLIENT-ID>&response_type=<RESPONSE-TYPE>&response_mode=<RESPONSE-MODE>
+```
+
+Note the following two parameters (`reponse_type` and `response_mode`) in the above request.
+
+<table>
+    <tr>
+        <th>
+            reponse_type
+        </th>
+        <td>
+            Informs WSO2 Identity Server (the authorization server) of the desired OAuth2 authorization processing flow, which also determines the type of authorization information that should be returned.</br></br>
+            Possible values:
+            <ul>
+                <li>
+                    <code>code</code>: Used if the <a href="../authorization-code-grant">authorization code flow</a> is implemented.
+                </li>
+                <li>
+                    <code>token</code>: Used if the <a href="../implicit-grant">implicit flow</a> is implemented.
+                </li>
+                <li>
+                    <code>code id_token</code>: Used if the <a href="../openid-connect-hybrid-flow">hybrid flow</a> is implemented.
+                </li>
+            </ul>
+        </td>
+    </tr>
+    <tr>
+        <th>
+            response_mode
+        </th>
+        <td>
+            Specifies how the authorization information should be returned to the client. By default, the following response modes are supported in WSO2 Identity Server: <code>query</code>, <code>fragment</code>, and <code>form_post</code>.</br></br>
+            In the default authorization flow (for default response modes), the authorization response parameters, including the authorization code and ID token, are sent as human-readable plain text to the redirect URL, as shown below.</br></br>
+            <code>https://{CLIENT_HOST}/redirects/redirect1?code={AUTH-CODE}&session_state={SESSION-STATE}</code></br></br>
```

Comment:
```suggestion
            <code>https://<CLIENT_HOST>/redirects/redirect1?code=<AUTH_CODE>&session_state=<SESSION_STATE></code></br></br>
```

Reply:
unfortunately, we can't render the `<` syntax inside an html table.
But let me try and find a way to make this more consistent.

Reply:
We can use &lt; instead of < and &gt; instead of >.
This should work:
<code>https://&lt;CLIENT_HOST&gt;/redirects/redirect1?code=&lt;AUTH-CODE&gt;&session_state=&lt;SESSION-STATE&gt;</code>

Reply:
this is done.